src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -892,14 +886,35 @@ void MeshForwarder::HandleMesh(uint8_t *               aFrame,
     }
     else if (meshHeader.GetHopsLeft() > 0)
     {
+        uint8_t priority = kDefaultMsgPriority;
+
         netif.GetMle().ResolveRoutingLoops(aMacSource.GetShort(), meshDest.GetShort());
 
         SuccessOrExit(error = CheckReachability(aFrame, aFrameLength, meshSource, meshDest));
 
         meshHeader.SetHopsLeft(meshHeader.GetHopsLeft() - 1);
         meshHeader.AppendTo(aFrame);
 
-        VerifyOrExit((message = GetInstance().GetMessagePool().New(Message::kType6lowpan, 0)) != NULL,
+#if OPENTHREAD_ENABLE_QOS
+        if (GetFramePriority(aFrame, aFrameLength, meshSource, meshDest, aLinkInfo, priority) != OT_ERROR_NONE)
+        {
+            priority = kDefaultMsgPriority;
+        }
+
+        if (aFrameLength - meshHeader.GetHeaderLength() >= 1 &&
+            reinterpret_cast<Lowpan::FragmentHeader *>(aFrame + meshHeader.GetHeaderLength())->IsFragmentHeader())
+        {
+            // An IPv6 packet may be fragmented into several fragments by 6Lowpan.
+            // The fragments, except the first fragment, don’t carry IPv6 header.
+            // The priority level comes from the DSCP value in IPv6 header. To get the
+            // priority level of the subsequent fragments, routers have to store the
+            // information of the fragments.
+            HandleFragment(aFrame + meshHeader.GetHeaderLength(), aFrameLength - meshHeader.GetHeaderLength(),
```

Comment:
The Thread specification does not currently require DSCP markings on Thread control traffic.  To properly handle traffic generated by existing Thread 1.1 implementations, we'll need to minimally inspect the UDP port to extract the priority.

Reply:
Do you mean that whether the DSCP value is set or not,  if the destination port of the message is the MLE UDP port,  we just set the priority of the message to High?